### PR TITLE
Kirkstone fixes

### PR DESCRIPTION
**Pull request by xzz53**
Update two patches and fix a few license-related QA warnings (mostly about `LICENSE = "BSD"`) for `noetic` and `humble`.

### CODE REVIEW

**Comment by krisklau** [comment]
Amazing! Thank you :+1:

File: meta-ros2-humble/recipes-bbappends/ament-lint/ament-cpplint_%.bbappend
Comment (robwoolley) [inline meta-ros2-humble/recipes-bbappends/ament-lint/ament-cpplint_%.bbappend:1]: ament-cpplint is under both Apache-2.0 and BSD

File: meta-ros2-humble/recipes-bbappends/orocos-kinematics-dynamics/orocos-kdl-vendor_%.bbappend
Comment (robwoolley) [inline meta-ros2-humble/recipes-bbappends/orocos-kinematics-dynamics/orocos-kdl-vendor_%.bbappend:1]: orocos_kdl_vendor appears to be dual licensed under Apache 2.0 (with its dependency orocos_kinematics_dynamics under LGPL-2.1-or-later)

File: meta-ros2-humble/recipes-bbappends/rosbag2/zstd-vendor_%.bbappend
Comment (robwoolley) [inline meta-ros2-humble/recipes-bbappends/rosbag2/zstd-vendor_%.bbappend:1]: The zstd-vendor package is Apache 2.0... the zstd library itself is BSD-3-Clause.  However, zstd is supplied by the system

File: meta-ros2-iron/recipes-bbappends/common-interfaces/actionlib-msgs_5.0.1-1.bbappend
Comment (robwoolley) [inline meta-ros2-iron/recipes-bbappends/common-interfaces/actionlib-msgs_5.0.1-1.bbappend:1]: This appears to be under Apache 2.0

**Review by robwoolley — COMMENTED** [review]
Thanks for submitting this.   Sorry that it took so long to provide feedback on.  I appreciate you going through and fixing up the licenses.   I left some comments on the licenses that I thought were incorrect.  Let me know what you think of the suggestions.

**Comment by xzz53** [comment]
Hi Rob, and thanks for the review! I've double checked and addressed all your suggestions. 

In fact, I have accumulated some more fixes along the same lines since this PR was submitted. I plan to make a separate PR for them, if you do not object.

**Comment by robwoolley** [comment]
Creating a separate PR sounds good.  I have merged this one into kirkstone-next.

**Pull request closed without merging**